src/main/java/core/basesyntax/model/FruitsShopServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.model;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class FruitsShopServiceImpl implements FruitShopService {
+    private final OperationStrategy operationStrategy;
+
+    public FruitsShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public Map<String, Integer> finalReport(List<FruitTransaction> parsed) {
+        if (parsed == null) {
+            throw new RuntimeException("Value can't be null");
+        }
+        parsed.forEach(t -> operationStrategy.get(t.getOperation()).operate(t));
+        return Storage.getMap();
+    }
```

Comment:
```suggestion
    @Override
    public void process(List<FruitTransaction> parsed) {
        if (parsed == null) {
            throw new RuntimeException("Value can't be null");
        }
        parsed.forEach(t -> operationStrategy.get(t.getOperation()).operate(t));
    }
```
in report service just access storage